File: relayer/package/services/ibc-types/mithril_client_state.go
package ibc_types

type MithrilHeightSchema struct {
	_              struct{} `cbor:",toarray"`
	RevisionNumber uint64
	RevisionHeight uint64
}

type MithrilProtocolParametersSchema struct {
	_    struct{} `cbor:",toarray"`
	K    uint64
	M    uint64
	PhiF *FractionSchema
}

type FractionSchema struct {
	_           struct{} `cbor:",toarray"`
	Numerator   uint64
	Denominator uint64
}

type MithrilClientStateSchema struct {
	_                  struct{} `cbor:",toarray"`
	ChainId            []byte
	LatestHeight       *MithrilHeightSchema
	FrozenHeight       *MithrilHeightSchema
	CurrentEpoch       uint64
	TrustingPeriod     uint64
	ProtocolParameters *MithrilProtocolParametersSchema
	UpgradePath        [][]byte
}

File: indexer/src/ibc-types/client/mithril_pb/MithrilClientState.ts
import {Data} from '../../plutus/data';
import {CardanoHeightSchema} from './CardanoHeight';
import {MithrilProtocolParametersSchema} from './MithrilProtocolParameters';

export const MithrilClientStateSchema = Data.Object({
  chain_id: Data.Bytes(),
  latest_height: Data.Nullable(CardanoHeightSchema),
  frozen_height: Data.Nullable(CardanoHeightSchema),
  current_epoch: Data.Integer(),
  trusting_period: Data.Integer(),
  protocol_parameters: Data.Nullable(MithrilProtocolParametersSchema),
  upgrade_path: Data.Array(Data.Bytes()),
});
export type MithrilClientState = Data.Static<typeof MithrilClientStateSchema>;
export const MithrilClientState = MithrilClientStateSchema as unknown as MithrilClientState;
